src/storage/mod.rs

```diff
@@ -20,7 +20,26 @@ pub mod storages;
 #[cfg(test)]
 mod tests;
 
-/// A trait that guarantees that mutable storage getter can be used on distinct indices from different threads.
+/// This is a marker trait which requires you to uphold the following guarantee:
+///
+/// > Multiple threads may call `get_mut()` with distinct indices without causing
+/// > undefined behavior.
+///
+/// This for example valid for `Vec`:
+///
+/// ```rust
+/// vec![1, 2, 3];
```

Comment:
Okay. I just thought it bit jarring to see the example just contain creation of `Vec`